src/bindings/python/src/pyopenvino/core/infer_request.hpp

```diff
@@ -17,58 +17,84 @@ typedef std::chrono::nanoseconds ns;
+        m_end_time = std::make_shared<Time::time_point>(Time::time_point{});
+
+        // Initialize InferRequest with default callback
+        if (set_default_callback) {
+            // Bump reference counter
+            auto end_time = m_end_time;
+            // Set standard callback which saves "end-time" for inference call
+            m_request.set_callback([end_time](std::exception_ptr exception_ptr) {
+                *end_time = Time::now();
+                try {
+                    if (exception_ptr) {
+                        std::rethrow_exception(exception_ptr);
+                    }
+                } catch (const std::exception& e) {
+                    throw ov::Exception("Caught exception: " + std::string(e.what()));
                 }
-            } catch (const std::exception& e) {
-                throw ov::Exception("Caught exception: " + std::string(e.what()));
-            }
-        });
+            });
+        }
     }
+
     // ~InferRequestWrapper() = default;
 
     std::vector<ov::Tensor> get_input_tensors() {
-        std::vector<ov::Tensor> tensors;
-        for (auto&& node : _inputs) {
-            tensors.push_back(_request.get_tensor(node));
-        }
-        return tensors;
+        return get_tensors_from(m_inputs);
     }
 
     std::vector<ov::Tensor> get_output_tensors() {
-        std::vector<ov::Tensor> tensors;
-        for (auto&& node : _outputs) {
-            tensors.push_back(_request.get_tensor(node));
-        }
-        return tensors;
+        return get_tensors_from(m_outputs);
     }
 
-    bool user_callback_defined = false;
-    py::object userdata;
-
     double get_latency() {
-        auto execTime = std::chrono::duration_cast<ns>(_end_time - _start_time);
+        auto execTime = std::chrono::duration_cast<ns>(*m_end_time - *m_start_time);
         return static_cast<double>(execTime.count()) * 0.000001;
     }
 
-    ov::InferRequest _request;
-    std::vector<ov::Output<const ov::Node>> _inputs;
-    std::vector<ov::Output<const ov::Node>> _outputs;
+    // Original ov::InferRequest class that is held by this wrapper
+    ov::InferRequest m_request;
+    // Inputs and Outputs inherrited from ov::CompiledModel
+    std::vector<ov::Output<const ov::Node>> m_inputs;
+    std::vector<ov::Output<const ov::Node>> m_outputs;
+    // A flag which is set when a user defines a custom callback on InferRequest
+    bool m_user_callback_defined = false;
+    // Data that is passed by user from Python->C++
+    py::object m_userdata;
+    // Times of inference's start and finish
+    std::shared_ptr<Time::time_point> m_start_time; // proposal: change to unique_ptr
+    std::shared_ptr<Time::time_point> m_end_time;
 
-    Time::time_point _start_time;
-    Time::time_point _end_time;
+private:
+    inline std::vector<ov::Tensor> get_tensors_from(const std::vector<ov::Output<const ov::Node>>& v) {
+        std::vector<ov::Tensor> tensors;
```

Comment:
Please add this line below the `tensors` vector declaration: `tensors.reserve(v.size())`

Reply:
Done